### ISSUE TITLE
Shouldn't RandomReverse be applied to both `im` and `gt`?

### ISSUE THREAD

**Issue by jcohenadad** [ivadomed/ivadomed#1280]
### Motivation for the feature

From the doc:
![image](https://user-images.githubusercontent.com/2482071/214969400-ed3f1002-bd9a-4220-8f4e-14d27c0bdd43.png)

It is suggested that RandomReverse is applied to `im` only. But if it was the case, the GT would not match. So, shouldn't we suggest by default to apply this transformation to both the image and the ground truth? Unless I am missing something.

**Comment by naga-karthik** [comment]
Yes, you're right! It has to be applied to both image and GT. For example, the [config file](https://github.com/ivadomed/model_seg_ms_mp2rage/blob/main/config/seg_lesion.json#L99) of the previous version of the mp2rage model applies it to both `im` and `gt`. Even for our ongoing centerline detection models, we are applying it for both images.

**Comment by jcohenadad** [comment]
OK, so in this case why don't we simply get rid of the 'applied_to', which introduces confusion and could lead to improper usage?

**Comment by naga-karthik** [comment]
Yes, I think that's a fair point! Conventionally, the user is left to specify which image is the transformation is applied to (this is also the case for Monai's transformations). I thought of adding a recommendation in the docs to apply it to both, but I think it's better to do it ourselves as you suggest. The confusion is indeed real and it would be better to not burden users with one more option among a million others!
